src/components/PageDefinitions.js

```diff
@@ -87,8 +84,21 @@ class PageDefinitions extends AbstractPageDefinitions {
     }
   }
 
+  // Get an array of definitions asynchronous, split them into 100 chunks and alert the user when they're all done
+  getDefinitionsAndNotify(definitions, message) {
+    const { dispatch, token } = this.props
+    const chunks = chunk(definitions, 100)
+    Promise.all(chunks.map(throat(10, chunk => dispatch(getDefinitionsAction(token, chunk)))))
```

Comment:
Using try/ catch and async/ await here is likely better. More in keeping with the rest of the code and safer.  As is, if some thing in the map fails before returning the promise, the catch handler will not catch it.  That failure would happen while committing the arg to all and would not reject a promise.